TheMovieDB/Model/MovieViewModel.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  MovieViewModel.swift
+//  TheMovieDB
+//
+//  Created by Intern on 13/07/23.
+//
+
+import Foundation
+
+protocol MovieViewModelProtocol: ObservableObject {
+    func loadMovies()
+    var currentPage: Int { get set }
+    var movieData: [Movies] { get }
+}
+
+class MovieViewModel: MovieViewModelProtocol {
+    @Published var currentPage = 1
+    @Published var movieData: [Movies] = [Movies]()
+    var networkDataSource: MovieDataSourceProtocol
+    init(networkDataSource: MovieDataSourceProtocol = MovieDataSource()){
+        self.networkDataSource = networkDataSource
+    }
+    //let dataSource = DummyDataSource()
+    func loadMovies() {
+       //movieData = dataSource.loadMovies(page: currentPage)
+        networkDataSource.getPopularMovies(page: currentPage) { queryResult in
```

Comment:
use [weak self] to avoid memory leak

resourcce to learn:
https://stablekernel.com/article/how-to-prevent-memory-leaks-in-swift-closures/#:~:text=A%20memory%20leak%20occurs%20in,memory%2C%20will%20cause%20a%20crash.